src/sql/workbench/parts/dataExplorer/electron-browser/nodeActions.contribution.ts

```diff
@@ -15,27 +15,83 @@ MenuRegistry.appendMenuItem(MenuId.DataExplorerContext, {
 		id: DISCONNECT_COMMAND_ID,
 		title: localize('disconnect', 'Disconnect')
 	},
-	when: NodeContextKey.IsConnected
+	when: ContextKeyExpr.and(
+		new ContextKeyDefinedExpr('isConnected'),
+		new ContextKeyEqualsExpr('isConnected', true))
 });
 
+// For Database nodes under a Server in the SQL Servers folder
 MenuRegistry.appendMenuItem(MenuId.DataExplorerContext, {
 	group: 'connection',
 	order: 2,
 	command: {
 		id: NEW_QUERY_COMMAND_ID,
 		title: localize('newQuery', 'New Query')
 	},
-	when: NodeContextKey.IsConnectable
+	when: ContextKeyExpr.and(
+		new ContextKeyDefinedExpr('isConnectable'),
+		new ContextKeyEqualsExpr('isConnectable', true),
+		new ContextKeyDefinedExpr('viewItem'),
+		new ContextKeyEqualsExpr('viewItem', 'Database'))
 });
 
+// For database nodes under the SQL Databases folder
+MenuRegistry.appendMenuItem(MenuId.DataExplorerContext, {
+	group: 'connection',
+	order: 2,
+	command: {
+		id: NEW_QUERY_COMMAND_ID,
+		title: localize('newQuery', 'New Query')
+	},
+	when: ContextKeyExpr.and(
+		new ContextKeyDefinedExpr('isConnectable'),
+		new ContextKeyEqualsExpr('isConnectable', true),
+		new ContextKeyDefinedExpr('viewItem'),
+		new ContextKeyEqualsExpr('viewItem', 'azure.resource.itemType.database'))
+});
+
+MenuRegistry.appendMenuItem(MenuId.DataExplorerContext, {
+	group: 'connection',
+	command: {
+		id: NEW_QUERY_COMMAND_ID,
+		title: localize('newQuery', 'New Query')
+	},
+	when: ContextKeyExpr.and(
+		new ContextKeyDefinedExpr('isConnectable'),
+		new ContextKeyEqualsExpr('isConnectable', true),
+		new ContextKeyDefinedExpr('viewItem'),
+		new ContextKeyEqualsExpr('viewItem', 'azure.resource.itemType.databaseServer'))
```

Comment:
This should be using regular expession rather than adding menu entries twice

Reply:
Yeah I'll make that change - although when I was looking at how people did this for VS Code menu extensions there were lots of suggestions to do it this way instead which is why I went this route. I assume the perf difference of not having to evaluate extra items is the big reason - is there anything else you know of (like incorrect behavior/duplicate entries) that would be why we shouldn't be doing it this way?

Reply:
Per my later comment, we don't have azure specific strings in core. using regex allows for the fix while not specializing for a specific extension. I.E CMS would want this behavior, but wouldn't get it unless we then hard coded another string for cms.

I'm a little confused on what you mean by perf difference. The perf of regular expression should be negligible. Do you have some link to what you are talking about with people talking about menu extensions? If you are referring to the checking if a context key is defined; I supposed doing a undefined check is faster than a reg ex, but for booleans (`isConnectable`) doing a undefined check then a true/false check is twice as slow than just doing a true/false check, since true/false check also does a undefined check at the same time.

Reply:
I was talking about perf of having separate menu registrations vs doing it as a regex. Without testing my gut feeling is that having the separate menu registrations is worse overall because each item would have to be evaluated separately every time you opened a menu - vs the regex which is just a single time. 

So that's why I'm doing the change to use a regex instead - I just wanted to see if you knew of any other reasons why we'd NOT want to do separate menu registrations. Because it seemed to work just fine functionality wise - just with the probable performance hit I mentioned above.